torchgeo/models/weights.py

```diff
@@ -0,0 +1,154 @@
+            "https://drive.google.com/file/d/1OrtPfG2wkO05bimstQ_T9Dza8z3zp8i-/"
+            "view?usp=sharing/B13_rn50_moco_0099_ckpt.pth"
+        ),
+        transforms=nn.Identity(),
+        meta={
+            "ssl_method": "moco",
+            "publication": "https://arxiv.org/abs/2211.07044",
+            "dataset": "SSL4EO-S12",
+            "repo": "https://github.com/zhu-xlab/SSL4EO-S12",
+        },
+    )
+    SENTINEL2_ALL_DINO = Weights(
+        url="https://drive.google.com/file/d/1iSHHp_cudPjZlshqWXVZj5TK74P32a2q",
+        transforms=nn.Identity(),
+        meta={
+            "ssl_method": "dino",
+            "publication": "https://arxiv.org/abs/2211.07044",
+            "dataset": "SSL4EO-S12",
+            "repo": "https://github.com/zhu-xlab/SSL4EO-S12",
+        },
+    )
+    DEFAULT = SENTINEL2_ALL_MOCO
+
+    def get_state_dict(self) -> Dict[str, Any]:
+        """Retrieve pretrained weights state_dict."""
+        root = os.path.join(os.getcwd(), "weights")
+        map_location = torch.device("cpu")
+
+        # need to define identifiers for each case
+        if self.meta["ssl_method"] == "dino":
+            filename = "resnet50_dino_sentinel_weights.pth"
+            ckpt = load_checkpoint_from_url(root, filename, self.url, map_location)
+            # https://github.com/zhu-xlab/SSL4EO-S12/blob/1a668f76fd46762a19780293675a6e23e5204e72/
+            # src/benchmark/transfer_classification/models/dino/utils.py#L92-L103
+            state_dict = ckpt["teacher"]
+            state_dict = {k.replace("module.", ""): v for k, v in state_dict.items()}
+            # remove `backbone.` prefix induced by multicrop wrapper
+            state_dict = {k.replace("backbone.", ""): v for k, v in state_dict.items()}
+
+        elif self.meta["ssl_method"] == "moco":
+            filename = "resnet50_moco_sentinel_weights.pth"
+            ckpt = load_checkpoint_from_url(root, filename, self.url, map_location)
+            # https://github.com/zhu-xlab/SSL4EO-S12/blob/d2868adfada65e40910bfcedfc49bc3b20df2248/
+            # src/benchmark/transfer_classification/linear_BE_moco.py#L248-L276
+            state_dict = ckpt["state_dict"]
+            for k in list(state_dict.keys()):
+                # retain only encoder up to before the embedding layer
+                if k.startswith("module.encoder_q") and not k.startswith(
+                    "module.encoder_q.fc"
+                ):
+                    # remove prefix
+                    state_dict[k[len("module.encoder_q.") :]] = state_dict[k]
+                # delete renamed or unused k
+                del state_dict[k]
+
+        elif self.meta["dataset"] == "imagenet":
+            filename = "resnet50_imagenet_rgb_weights.pth"
+            state_dict = load_checkpoint_from_url(
+                root, filename, self.url, map_location
+            )
+
+        elif self.meta["dataset"] == "millionaid":
+            try:
+                import yacs  # noqa: F401
```

Comment:
What is this and why is it necessary?

Reply:
it's necessary for this set of weights, otherwise 'load_checkpoint' will thrown an error saying that this library is missing. so it seems the authors of those weights used [yacs](https://github.com/rbgirshick/yacs).

Reply:
I'm probably going to remove this specific set of weights. We can add it later and see if there's an easier way to load it. Maybe we can repackage it ourselves and upload it somewhere so yacs isn't required.